### ISSUE TITLE
Edit a Ticket in Frontend

### ISSUE THREAD

**Issue by Chris-Boe** [codeforpdx/dwellingly-app#527]
**Purpose**
What will this feature provide?
Ability to edit some details in the Ticket modal

**Proposed Solution**
If you have an idea on how to implement this feature, please share!
* Remove the speech bubble in the header of the modal
* Add an edit Pencil button to the right of the modal header

**Acceptance Criteria**
Please list the requirements the implementation of this feature should meet.
- [ ] Allows editing of the Sender, Tenant, Urgency, and Assignee (when #525 is done)
- [ ] When in edit mode, a Save button will be shown above the page break at the bottom of the general details section
- [ ] Changes should be saved to backend when save button is shown

**Mockup Screenshot or Page Number**
Paste a screenshot of the feature from the FIGMA mockups or cite the page number.
![image](https://user-images.githubusercontent.com/8600792/107600889-eccf0e80-6bd9-11eb-949b-ab4c5067dab2.png)

**Additional context**

**Comment by Protopaco** [comment]
My plan for this ticket:

- refactor current ticket title into an independent component "TicketModalDetails" 
- create "EditTicketModalDetails" component that will have a form, text fields, and save button that replicates styling of "TicketModalDetails"
- add pencil icon that will toggle between "TicketModalDetails" and "EditTicketModalDetails"
- create handleEditDetails function that will send updates to the backend and update client data with changes

Let me know if you think there is a better way to do this.  Thanks!

**Comment by Protopaco** [comment]
> * create "EditTicketModalDetails" component that will have a form, text fields, and save button that replicates styling of "TicketModalDetails"

I just realized that of course a text field isn't going to work.  They are going to have to be pull down menus that reference available tenants, property managers, etc.   Not a big deal, I will adjust the plan accorindly.

**Comment by Protopaco** [comment]
Please disregard my previous commit.  I got past my roadblock, and I'm making good progress again.  I'm hoping to have a working model done before the meeting tomorrow.

**Comment by Protopaco** [comment]
I just pushed a new commit that I think gets us 95% of the way there.  Included changes:

- user may edit Sender, Tenant, Assigned Manager, and Urgency on a ticket
- removes blue balloon, and replaces it with an "edit pen"
- adds blue "save" button, that sends the information to the server and updates all tickets.
- adds a "SearchableDropDown" component that allows the user to search through the available choices

I had a couple of questions before we can close this ticket.  I looked for the answers, but I couldn't find them.

- Who is eligible to "Send" a ticket?  For this I included Staff and Admin.  It is an easy change
- What are the urgency levels for a ticket?  I included LOW, MEDIUM, & HIGH.  
- The pen icon looks a little awkward in its current placement.  I played around with it for far too long, but couldn't get a fit I liked.  Would love to get a second opinion on that styling.

One other note, I tried using the SearchPanel component that was used in "addTenant" but it I couldn't find a way to make it work in the confined size of the ticket.   I built the "SearchableDropDown" component to provide most of the same functionality in a smaller package.

Thank you!
